Automattic/jetpack
# Compatibility fix with the DSP checkout endpoint for the Blaze Dashboard

Fixes a 404 error we are getting in the Blaze Dashboard checkout endpoint when creating a Blaze campaign.

## Proposed changes:
* Add the mapping for the missing `wordads/dsp/api/v1/wpcom/checkout` endpoint inside the Blaze package
* I am also removing a duplicate method that we had to check permissions and change all the endpoints to use the new one. (We want to check that the user is connected and manage permissions in all the endpoints).
* I increased the timeout for the heavy operations on the DSP to 20. That would be the checkout method and the campaign creation post methods (e.g., uploading the creative images).

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






On a self-hosted site, enable the Blaze Dashboard with `add_filter( 'jetpack_blaze_dashboard_enable', '__return_true' );`

* Go to Tools->Advertising
* Click on the Promote button for one of your posts
* Fill all the necessary information (if you got to the express flow, you need to click the Make changes button).
* If you already have credit card information saved, click the `Add new card` link
* Input all the data requested on the payment form In the payment form
* Click **Submit campaign** button
* Verify that the call to the checkout endpoint finishes correctly, and that the campaign was created ok.
![Screenshot 2023-07-07 at 14 38 10](https://github.com/Automattic/jetpack/assets/1258162/02da8a56-15ee-4709-a1bd-134ee1c61e87)

You can then go to your campaigns, enter the details and cancel the campaign by clicking the **Cancel campaign** button.
